### DISCUSSION TITLE
Enable REPLACE behaviour using appropriate MergeTree engine

### DISCUSSION THREAD

**Discussion by ana-singla** [ClickHouse/ClickHouse#70583] [Q&A]
I want to push data to Clickhouse in a manner that the old data always gets replaced with the new data. 
Example:
current data and state of the table:
```
{"location_id": "A", "area": "area-a", "capacity": 1}
{"location_id": "B", "area": "area-b", "capacity": 2}
{"location_id": "C", "area": "area-c", "capacity": 3}
```
new data:
```
{"location_id": "B", "area": "area-b", "capacity": 20}
{"location_id": "C", "area": "area-c", "capacity": 30}
{"location_id": "D", "area": "area-d", "capacity": 40}
```

Final state of the table:
```
{"location_id": "B", "area": "area-b", "capacity": 20}
{"location_id": "C", "area": "area-c", "capacity": 30}
{"location_id": "D", "area": "area-d", "capacity": 40}
```

We're using materialized views for materializing the final state of the data which will be queried.
I tried using `VersionedCollapsingMergeTree(sign, version)` where I set sign as `1` always so the engine only considers the version column. I suppose this is not how this engine is to be used. I cannot insert the old columns with sign as `-1`. 
Is there any way to implement this functionality either with `VersionedCollapsingMergeTree` or any other way?

**Comment by SaltTan** [comment]
Have you tried ReplacingMergeTree?
https://clickhouse.com/docs/en/guides/replacing-merge-tree

**Reply by ana-singla** [reply]
ReplacingMergeTree is about UPSERT behaviour, it will update the values for my primary key set. I want all the previous data to be removed. We cannot truncate a materialized view, we can only drop it

**Comment by SaltTan** [comment]
Maybe Refreshable Materialized View then 
https://clickhouse.com/docs/en/materialized-view/refreshable-materialized-view
